memory/includes/phys_mem.hpp

```diff
@@ -19,18 +16,21 @@ class PhysMem final {
     [[nodiscard]] static PhysMem *CreatePhysMem(uint64_t total_size);
     static bool Destroy(PhysMem *phys_mem);
 
-    template <bool AllocPageIfNeeded>
-    uint8_t *GetAddr(uint64_t page_id, uint64_t offset);
-    Page *GetPage(uint64_t page_id);
+    bool Read(uintptr_t paddr, size_t size, void *value);
+    bool Write(uintptr_t paddr, size_t size, void *value);
+    uint64_t GetEmptyPageNumber() const;
+    bool InitPage(uintptr_t paddr);
     bool AtOnePage(uint64_t offset, uint64_t length) const;
-    std::pair<uint64_t, uint64_t> NextAfterLastOccupiedByte() const;
+    uint8_t *GetMemPointer() const;
 
 private:
     PhysMem(uint64_t size);
-    ~PhysMem() = default;
+    ~PhysMem();
 
     uint64_t total_size_;
-    std::vector<Page> pages_;
+    // TODO(Mirageinvo): return Page class?
```

Comment:
Remove TODO, it's better solution then std::vector<Page>

Reply:
Done